html/cross-origin-embedder-policy/resources/worker-owner-frame.html

```diff
@@ -0,0 +1,29 @@
+<!doctype html>
+<script>
+function startWorkerAndObserveReports(worker_url, wait_for_report) {
+  const worker = new Worker(worker_url);
+  const result_promise = new Promise(resolve => {
+    worker.onmessage = _ => resolve('success');
+    worker.onerror = _ => resolve('error');
+  });
+  worker.postMessage("postMessage('reply to frame from worker');");
+
+  const report_promise = new Promise(resolve => {
+    const observer = new ReportingObserver(reports => {
+      observer.disconnect();
+      resolve(reports.map(r => r.toJSON()));
+    });
+    observer.observe();
+  });
+
+  if (wait_for_report) {
+    Promise.all([result_promise, report_promise]).then(results => {
+      parent.postMessage(results[1]);
+    });
+  } else {
+    result_promise.then(result => {
+      parent.postMessage([]);
```

Comment:
@d0iasm so this looks like it will pass the test even if a report is created, right? Doesn't this make the length 0 tests not actually verify that no reports are created?